File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int i=0, j=height.size()-1;
        int water = 0;
        while (i < j){
            water = max(water, min(height[i], height[j]) * (j-i));
            if (height[i] <= height[j]){
                i+=1;
            } else {
                j-=1;
            }
        }

        return water;
    }
};

File: 11-container-with-most-water/container-with-most-water.go
func maxArea(height []int) int {
    i, j := 0, len(height)-1
    max_water := 0
    for i < j {
        l := min(height[i], height[j])
        b := j - i

        water := l * b
        max_water = max(max_water, water)

        if height[i] < height[j] {
            i++
        } else {
            j--
        }
    }
    return max_water
}